pyctcdecode/alphabet.py

```diff
@@ -44,48 +81,39 @@ def _convert_bpe_format(token: str) -> str:
         return BPE_CHAR + token
 
 
-def _normalize_bpe_alphabet(
-    label_list: List[str],
-    unk_token_idx: Optional[int] = None,
-    ctc_token_idx: Optional[int] = None,
-) -> List[str]:
+def _normalize_bpe_alphabet(labels: List[str]) -> List[str]:
     """Normalize alphabet for bpe decoder."""
-    if ctc_token_idx is None:
-        ctc_token_idx = _get_ctc_index(label_list)
-    # create copy
-    clean_labels = label_list[:]
-    # there are two common formats for BPE vocabulary
-    # 1) where ▁ indicates a space (this is the main format we use)
-    if any([s[:1] == BPE_CHAR and len(s) > 1 for s in clean_labels]):
-        # verify unk token and make sure it is consistently represented as ▁⁇▁
-        if unk_token_idx is None and clean_labels[0] in ("<unk>", UNK_BPE_CHAR):
-            unk_token_idx = 0
-        else:
-            raise ValueError(
-                "First token in vocab for BPE should be '▁⁇▁' or specify unk_token_idx."
-            )
-        clean_labels[unk_token_idx] = UNK_BPE_CHAR
-    # 2) where ## indicates continuation of a token (note: also contains the single token: ▁)
-    elif any([s[:2] == "##" for s in clean_labels]):
-        # convert to standard format 1)
-        clean_labels = [_convert_bpe_format(c) for c in clean_labels]
-        # add unk token if needed
-        if clean_labels[0] in ("<unk>", UNK_BPE_CHAR):
-            clean_labels[0] = UNK_BPE_CHAR
-        else:
-            clean_labels = [UNK_BPE_CHAR] + clean_labels
-            ctc_token_idx += 1
-    else:
+    normalized_labels = labels[:]
+    # if BPE is of style '##' then convert it
+    if any([s.startswith("##") for s in labels]):
+        normalized_labels = [_convert_bpe_token_style(c) for c in normalized_labels]
+    # raise if we don't end up in style '▁'
+    if not any([s.startswith(BPE_CHAR) for s in labels]):
         raise ValueError(
             "Unknown BPE format for vocabulary. Supported formats are 1) ▁ for indicating a"
             " space and 2) ## for continuation of a word."
         )
-    # specify ctc blank token
-    if ctc_token_idx == -1:
-        clean_labels.append("")
-    else:
-        clean_labels[ctc_token_idx] = ""
-    return clean_labels
+    # substitute unk
+    for n, label in enumerate(normalized_labels):
+        if label.lower() in ("unk", "<unk>", "⁇"):
+            logger.info("Found %s in vocabulary, substituting with %s." % (label, UNK_BPE_CHAR))
+            normalized_labels[n] = UNK_BPE_CHAR
+    if UNK_BPE_CHAR not in normalized_labels:
+        logger.info("UNK not found in labels, prepending to beginning.")
+        normalized_labels = [UNK_BPE_CHAR] + normalized_labels
```

Comment:
are we allowed to do this?  could we be making an assumption about the acoustic model and logit structure that might not be true, in which case we'd be throwing the whole thing off?